client/my-sites/themes/events/theme-showcase-tracks.ts

```diff
@@ -0,0 +1,77 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { property, snakeCase } from 'lodash';
+
+type ThemeSearchQuery = {
+	search: string;
+	tier: string;
+	filter: string;
+	page: number;
+	number: number;
+};
+
+type Theme = {
+	id: string;
+};
+
+enum ThemeShowcaseEventType {
+	ThemeClick = 'calypso_themeshowcase_theme_click',
+	ThemeStyleVariationClick = 'calypso_themeshowcase_theme_style_variation_click',
+	ThemeStyleVariationMoreClick = 'calypso_themeshowcase_theme_style_variation_more_click',
+}
+
+const getThemeShowcaseEventRecorder = (
+	query: ThemeSearchQuery,
+	themes: Array< Theme >,
+	getThemeType: ( themeId: string ) => string,
+	isActiveTheme: ( themeId: string ) => boolean,
+	defaultCollectionId: string | null = null,
+	defaultCollectionIndex: number | null = null
+) => {
+	const getEventRecorder =
+		( event: ThemeShowcaseEventType ) =>
+		(
+			themeId: string,
+			themePosition: number,
+			action: string = '',
+			styleVariation: string = '@theme',
+			collectionId: string | null = null,
+			collectionIndex: number | null = null
+		) => {
+			const adjustedPosition = themePosition + 1;
+			const isCollection = null !== defaultCollectionId || null !== collectionIndex;
+
+			recordTracksEvent( event, {
+				search: query?.search,
+				tier: query?.tier,
+				filter: query?.filter,
```

Comment:
Before we were recording the `filterString` which isn't always available and depends on the client building it based on the `query` prop. So I just changed this to show the actual query instead which has the same information (actually more since the tier wasn't included in the filter string).

Reply:
Maybe I'm misunderstanding this but, since we are extracting an existing functionality, let's not change its behaviour in the same PR.

I'm fine adding new props for the collections, but not about changing existing ones.

If it's necessary to change them to make it work with collections, then I'd rather split this task in 2:
1. Extract the tracking and apply it to existing use cases.
2. Apply it to the new (collection) use cases.

Besides, what if some funnels (or other data analyses) rely on the current `filter` expected values?

It's generally safer to add a new property (and describe the difference clearly in the event details) — but then again, I'm not exactly sure what is changing here — which is kind of my point. 😅 

I've taken a screen of the `calypso_themeshowcase_theme_click` prop on prod vs dev (clicking on the Fewer theme from the `/filter/portfolio` URL):

| Prod | Dev |
|--------|--------|
| <img width="303" alt="Screenshot 2023-10-26 at 12 06 37" src="https://github.com/Automattic/wp-calypso/assets/2070010/1c9bd278-ebd0-4c6a-9acd-df81bef4b551"> | <img width="303" alt="Screenshot 2023-10-26 at 12 06 53" src="https://github.com/Automattic/wp-calypso/assets/2070010/527ca51b-a4e0-457b-8abb-460f1081e652"> | 

We are removing several props (e.g. `search_taxonomies`, `search_term`), and replacing them with others (e.g. `filter`, `search`, `tier`).

What about keeping both for now?

Reply:
> What about keeping both for now?

I would need to drill down the function that calculates `search_taxonomies` and `search_terms`. Both of those fields are kinda of the same and don't include the `tier` of searches. I think that the current implementation is a bit lacking since it's more complex than my proposal and includes less information than my proposal. 

But I can restore the search taxonomies and rank for the non-collection events and in a follow-up remove it + the duplicated events.

So to recap, I'll restore those props, then in a follow-up speak with lego about them and the duplicated events.

Reply:
Thanks! 🙇‍♂️ 

> I think that the current implementation is a bit lacking since it's more complex than my proposal and includes less information than my proposal.

I agree!
It's just to make our changes more predictable, and hopefully avoid regressions on code-adjacent tasks, like data analysis.